xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp

```diff
@@ -119,37 +119,39 @@ bool CRendererMediaCodecSurface::Supports(ERENDERFEATURE feature)
 void CRendererMediaCodecSurface::RenderUpdate(int index, int index2, bool clear, unsigned int flags, unsigned int alpha)
 {
   CXBMCApp::get()->WaitVSync(100);
-  ManageRenderArea();
   m_bConfigured = true;
-}
-
-void CRendererMediaCodecSurface::ReorderDrawPoints()
-{
-  CBaseRenderer::ReorderDrawPoints();
 
   // this hack is needed to get the 2D mode of a 3D movie going
   RENDER_STEREO_MODE stereo_mode = g_graphicsContext.GetStereoMode();
   if (stereo_mode)
     g_graphicsContext.SetStereoView(RENDER_STEREO_VIEW_LEFT);
 
+  ManageRenderArea();
```

Comment:
This is quite CPU consuming, so moving it to this place where it is called every frame could not be the best option.
Why has this calculation be done for every frame?

Background: 4K on non Shield devices (much less CPU power) is really on limit, so we should try to move things out of the frame loop whenever possible.

Reply:
Maybe the diff fools you, but it's exactly the same place as before: 
https://github.com/koying/xbmc/blob/0ab8b2eb7f49364b747ad0e162a0127ffc298704/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp#L129

https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp#L122